pkg/operator/ceph/cluster/mgr/mgr.go

```diff
@@ -339,9 +340,57 @@ func startModuleConfiguration(description string, configureModules func() error)
 }
 
 // Ceph docs about the prometheus module: http://docs.ceph.com/docs/master/mgr/prometheus/
-func (c *Cluster) enablePrometheusModule() error {
-	if err := cephclient.MgrEnableModule(c.context, c.clusterInfo, PrometheusModuleName, true); err != nil {
-		return errors.Wrap(err, "failed to enable mgr prometheus module")
+func (c *Cluster) configurePrometheusModule() error {
+	if c.spec.Monitoring.Enabled {
```

Comment:
@galexrt @avanthakkar If monitoring is not enabled in the cluster CR, the proposed behavior is that the prometheus mgr module would also be disabled. Is there any reason I'm missing why the prometheus mgr might need to remain enabled in that case? In the past Rook has just always enabled the prometheus mgr module, so please point out if I'm forgetting some reason to leave it enabled always.

Reply:
I don't think the prometheus module is required for anything else than monitoring the cluster using a Prometheus.